QuizApp/NetworkService.swift

```diff
@@ -0,0 +1,190 @@
+//
+//  NetworkService.swift
+//  QuizApp
+//
+//  Created by Marin on 10.05.2021..
+//
+
+import UIKit
+
+class NetworkService: NetworkServiceProtocol {
+    private let baseURL: String! = "iosquiz.herokuapp.com"
+    private let apiToken: String! = "api_token"
+    private let userId: String! = "user_id"
+    private let userDefaults = UserDefaults.standard
+    
+    func login(username: String, password: String, completionHandler: @escaping (Result<Bool, RequestError>) -> Void) {
+        let session = URLSession(configuration: URLSessionConfiguration.default)
+        var params: [String: String] = [:]
+        params["username"] = username
+        params["password"] = password
+        let urlComponent = addURLComponent(path: "/api/session", params: params)
+        guard let url = urlComponent.url else {
+            completionHandler(.failure(.serverError))
+            return
+        }
+        
+        var urlRequest = URLRequest(url: url)
+        urlRequest.httpMethod = "POST"
+        
+        session.dataTask(with: urlRequest){
+            data, response, error in
+            if error != nil {
+                completionHandler(.failure(.serverError))
+                return
+            }
+            
+            guard let response = response as? HTTPURLResponse else {
+                completionHandler(.failure(.serverError))
+                return
+            }
+            
+            if !(200...299).contains(response.statusCode) {
+                completionHandler(.failure(.serverError))
+                return
+            }
+        
+            guard let data = data else {
+                completionHandler(.failure(.noDataError))
+                return
+            }
+            
+            guard let loginData: LoginData = try? JSONDecoder().decode(LoginData.self, from: data) else {
+                completionHandler(.failure(.decodingError))
+                return
+            }
+            
+            self.userDefaults.set(loginData.token, forKey: self.apiToken)
+            self.userDefaults.set(loginData.userId, forKey: self.userId)
+            
+            completionHandler(.success(true))
```

Comment:
When will it be .success(false)? Will it ever? If not, you don’t need a bool as a success parameter